### PR TITLE
Status menu code quality improvements

### PR DESCRIPTION
**Pull request by igorkulman**
Mostly removing repeated boilerplate code and structure cleanup.

### CODE REVIEW

**Comment by igorkulman** [comment]
Ready for review

**Comment by williambj1** [comment]
@igorkulman Thank you very much for all those pull requests that significantly improved code quality.

I've just merged @Goshin's PR #33, could you please resolve the conflicts?

Thanks in advance!

**Comment by williambj1** [comment]
Thx

**Comment by igorkulman** [comment]
I fixed the conflict but it probably better be checked by @Goshin if I resolved everything correctly.

File: HeliPort/Appearance/StatusMenu.swift [hunk 1]
```diff
@@ -43,7 +49,7 @@ final class StatusMenu: NSMenu, NSMenuDelegate {
                 StatusBarIcon.connected()
             default:
                 StatusBarIcon.off()
-            }
+            }
```
Comment (Goshin) [inline HeliPort/Appearance/StatusMenu.swift:52]: trailing whitespaces

File: HeliPort/Appearance/StatusMenu.swift [hunk 2]
```diff
@@ -17,76 +17,52 @@ import Foundation
...
-            case ITL80211_S_INIT.rawValue:
-                statusText = "Wi-Fi: On"
+            case ITL80211_S_INIT:
                 StatusBarIcon.disconnected()
-            case ITL80211_S_SCAN.rawValue:
-                statusText = "Wi-Fi: Looking for Networks..."
-            case ITL80211_S_AUTH.rawValue, ITL80211_S_ASSOC.rawValue:
-                statusText = "Wi-Fi: Connecting"
+            case ITL80211_S_SCAN, ITL80211_S_AUTH, ITL80211_S_ASSOC:
                 StatusBarIcon.connecting()
```
Comment (Goshin) [inline HeliPort/Appearance/StatusMenu.swift:47]: no need to `StatusBarIcon.connecting()` for `ITL80211_S_SCAN`.

File: HeliPort/Appearance/StatusMenu.swift [hunk 3]
```diff
@@ -97,28 +76,40 @@ class StatusMenu: NSMenu, NSMenuDelegate {
...
             self.isNetworkListEmpty = !newState
         }
     }
 
-    override init(title: String) {
-        super.init(title: title)
+    // - MARK: Menu items
+
+    private let statusItem = NSMenuItem(title: NSLocalizedString("Wi-Fi: Status unavailable", comment: ""))
+    private let switchItem = NSMenuItem(title: NSLocalizedString("Turn Wi-Fi Off", comment: ""))
```
Comment (williambj1) [inline HeliPort/Appearance/StatusMenu.swift:104]: `switchItem` should be clickable

File: HeliPort/Appearance/StatusMenu.swift [hunk 4]
```diff
@@ -312,9 +312,7 @@ final class StatusMenu: NSMenu, NSMenuDelegate {
             get_80211_state(&status)
 
             DispatchQueue.main.async {
-                if get_power_ret {
-                    self.isNetworkEnabled = powerState
-                }
+                self.isNetworkEnabled = get_power_ret
```
Comment (Goshin) [inline HeliPort/Appearance/StatusMenu.swift:315]: `get_power_ret` only represents the result of the API call (succeeded or failed), the obtained power state (on or off) is in `powerState` var. The switch will be updated when the power state is successfully obtained here.

**Pull request merged**